webdictionary/src/main/java/com/eis0/kademlianetwork/TableUpdateIterator.java

```diff
@@ -0,0 +1,79 @@
+package com.eis0.kademlianetwork;
+
+import com.eis.smslibrary.SMSPeer;
+import com.eis0.kademlia.KademliaId;
+import com.eis0.kademlia.SMSKademliaNode;
+import com.eis0.kademlia.SMSKademliaRoutingTable;
+
+/**
+ * Iterator for the execution of the update-table algorithm.
+ * I'm using an iterator because in the future where we have more KademliaNetwork to keep
+ * track of, this iterator is specific for the single network, so we can have more than once
+ * running on different tables at the same time
+ *
+ * @author Marco Cognolato
+ */
+public class TableUpdateIterator {
+    private int currentCount;
+    private int maxCount;
+    private KademliaId netId;
+    private SMSKademliaRoutingTable table;
+    private SMSPeer netPeer;
+
+    public TableUpdateIterator(int maxCount, KademliaId netId, SMSKademliaRoutingTable table, SMSPeer netPeer) {
```

Comment:
Add some checks about `maxCount`, `netId` (if not already done in the `KademliaId` class), `table` and `netPeer`:

- What happens if `maxCount `is 0 or negative?
- Is `netId `always a valid ID?
- What happens if `table` is null? You could catch this "event" in advance and avoid useless code execution.
- Same thing for `netPeer`.

Reply:
Awesome suggestion!
Working on it 👍